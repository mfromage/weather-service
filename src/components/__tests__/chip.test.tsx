import Chip from '@/components/chip';
import {fireEvent, render} from '@testing-library/react-native';

describe('Chip', () => {
  const setup = (props = {}) => {
    const testId = 'chip-test-id';
    const defaultProps = {
      title: 'Hello world',
      item: {id: testId},
      onPress: jest.fn(),
    };
    const utils = render(<Chip {...defaultProps} {...props} testID={testId} />);
    const chip = utils.getByTestId(testId);
    return {...utils, chip};
  };
  it('render with given props', () => {
    const title = 'Hello world';
    const {getByText} = setup({title});

    expect(getByText(title)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const {toJSON} = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('call onPress with given item', () => {
    const item = {id: 1, name: 'hello'};
    const onPress = jest.fn();
    const {chip} = setup({item, onPress});

    fireEvent.press(chip);
    expect(onPress).toHaveBeenCalledWith(item);
  });

  it('not call onPress if selected', () => {
    const onPress = jest.fn();
    const {chip} = setup();

    fireEvent.press(chip);
    expect(onPress).not.toHaveBeenCalled();
  });
});
