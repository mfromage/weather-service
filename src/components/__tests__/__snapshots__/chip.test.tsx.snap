// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip matches the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="chip-test-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "blue",
          "borderRadius": 16,
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 12,
          "paddingVertical": 0,
        },
        false,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "blue",
            "fontSize": 12,
          },
          false,
        ]
      }
    >
      Hello world
    </Text>
  </View>
</View>
`;
